pkg/controller/podautoscaler/horizontal.go

```diff
@@ -396,6 +396,9 @@ func (a *HorizontalController) reconcileAutoscaler(hpav1Shared *autoscalingv1.Ho
 	} else if currentReplicas > hpa.Spec.MaxReplicas {
 		rescaleReason = "Current number of replicas above Spec.MaxReplicas"
 		desiredReplicas = hpa.Spec.MaxReplicas
+	} else if currentReplicas == hpa.Spec.MaxReplicas {
+		rescale = false
+		setCondition(hpa, autoscalingv2.ScalingActive, v1.ConditionFalse, "ScalingNoNeed", "no need to scale since the replica count equals maximum limit")
```

Comment:
is this making a pod set non-scalable once its number of replicas reached its max?
even a scale-in operation becomes not possible now?

Reply:
Also a little bit concerned about adding new Condition as literal strings in an adhoc way.